src/targets/maven.ts

```diff
@@ -0,0 +1,329 @@
+      checkEnvForPrerequisite({ name });
+      return {
+        name,
+        value: process.env[name],
+      };
+    });
+    return this.reduceConfig(secrets);
+  }
+
+  private reduceConfig(
+    config: { name: string; value: string | undefined }[]
+  ): Record<string, string> {
+    return config.reduce((prev, current) => {
+      return {
+        ...prev,
+        [current.name]: current.value,
+      };
+    }, {});
+  }
+
+  private getOuterTargetSettings(): Record<TargetSettingType, string> {
+    const settings = targetOptions.map(setting => {
+      if (!this.config[setting]) {
+        throw new ConfigurationError(
+          `Required configuration ${setting} not found in configuration file. ` +
+            `See the documentation for more details.`
+        );
+      }
+      return {
+        name: setting,
+        value: this.config[setting],
+      };
+    });
+    return this.reduceConfig(settings);
+  }
+
+  private getAndroidSettings(): AndroidFields {
+    if (
+      !this.config.android.distDirRegex ||
+      !this.config.android.fileReplaceeRegex ||
+      !this.config.android.fileReplacerStr
+    ) {
+      throw new ConfigurationError(
+        'Required Android configuration not found in configuration file. ' +
+          'See the documentation for more details.'
+      );
+    }
+
+    return {
+      android: {
+        distDirRegex: stringToRegexp(this.config.android.distDirRegex),
+        fileReplaceeRegex: stringToRegexp(
+          this.config.android.fileReplaceeRegex
+        ),
+        fileReplacerStr: this.config.android.fileReplacerStr,
+      },
+    };
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   * If there's required software missing, raises an error.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug(
+      'Checking if Maven CLI is available: ',
+      this.mavenConfig.mavenCliPath
+    );
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
+    this.logger.debug(
+      'Checking if Gradle CLI is available: ',
+      this.mavenConfig.gradleCliPath
+    );
+    checkExecutableIsPresent(this.mavenConfig.gradleCliPath);
+    this.logger.debug(
+      'Checking if GPG is available: ',
+      this.mavenConfig.gradleCliPath
+    );
+    checkExecutableIsPresent('gpg');
+  }
+
+  /**
+   * Publishes current Java and Android distributions.
+   * @param version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revison: string): Promise<void> {
+    await this.createUserGradlePropsFile();
+    await this.upload(revison);
+
+    // Maven central is very flaky, so retrying with an exponential delay in
+    // in case it fails.
+    await retrySpawnProcess(this.mavenConfig.gradleCliPath, [
+      'closeAndReleaseRepository',
+    ]);
```

Comment:
right now a Gradle plugin does it (close and promote the releases): https://github.com/vanniktech/gradle-maven-publish-plugin/tree/master/src/main/kotlin/com/vanniktech/maven/publish/nexus

if we do those HTTP requests directly via Craft, we could remove the Gradle plugin for good.

Reply:
There's no support for the Gradle plugin yet, it will be supported at a later stage. This is targeting https://github.com/getsentry/sentry-java.

Reply:
that's not really related to our sentry android gradle plugin, but rather the graadle plugin that calls `closeAndReleaseRepository`.
if we replicate what that plugin does via craft, we can eliminate the usage of 3rd party plugins to release our things

Reply:
the maven cli does not have the ability to close the repo and promote the releases on sonatype, thats why we use this plugin, otherwise, we'd need to do it manually.

Reply:
Added a TODO, we can address this later.